# Machine Learning Engineer Nanodegree
## Capstone Proposal and Capstone Project

### Overview
The host, Memorial Sloan Kettering Cancer Center (MSKCC), has been maintaining the database OncoKB for the purpose of knowledge sharing of mutation effects among oncologist. Currently this interpretation of genetic mutations is being done manually. This is a very time-consuming task where a clinical pathologist has to manually review and classify every single genetic mutation based on evidence from text-based clinical literature. And due to my combined interest in biology and data science, I am interested in using my expertise to develop a machine learning algorithm that, using an expert-annotated knowledge base as a baseline, automatically classifies genetic variations.

This is my very first competition since I started my journey of data science the middle of last year. And it has become a unforgetful fast learning experience for me. The rigorious background research, heated forum discussions, sometimes unpredictable model responses, and  the hectic and exhuasting feature engineering implementations, have altoghether given me a read taste of the machine learning implementation on an actual need. The competition ends at October 2nd, 2017. I am very happy to see the top 5% scores in my novice trial as reward of the hardwork I have put. If you are interested, you can check my `doumentation` for more details about the solution.

I found it’s rather interesting to see how machine learning can help in routine biological study. Actually, there are tons of repetitive work in a junior researcher’s career. For me, as my project is using biophysical method to study the structural and functional characteristics of a pathological protein, I am exposed to lots of NMR data every day. It always takes weeks or even months to manually assign amino acid fingerprint son HSQC to its primary structure by comparing their chemical shift with empirical values, which is totally boring and mechanic. Therefore, I am very interested in applying my knowledge and skills I learned here to change this situation.

**Note:**
- The mentioned kaggle competition is [here](https://www.kaggle.com/c/msk-redefining-cancer-treatment).
- All the files mentioned could be found in `data` folder or accessed [here](https://www.kaggle.com/c/msk-redefining-cancer-treatment/data).  
- My kaggle profile is [here](https://www.kaggle.com/lu1993).

### Software and Libararies
- [Python 2.7](https://www.python.org/download/releases/2.7/)
- [NumPy](http://www.numpy.org/)
- [pandas](http://pandas.pydata.org/)
- [scikit-learn](http://scikit-learn.org/stable/)
- [matplotlib](http://matplotlib.org/)
- [Seaborn](https://seaborn.pydata.org)
- [Gensim (word2vec)](https://radimrehurek.com/gensim/)
- [XGBOOST](https://xgboost.readthedocs.io/en/latest/)
- [Entrez-biopython](http://biopython.org/DIST/docs/api/Bio.Entrez-module.html)

### Skills Used:
- Research and investigate a real-world problem of interest.
- Accurately apply specific machine learning algorithms and techniques.
- Properly analyze and visualize your data and results for validity.
- Quickly pick up unfamiliar libraries and techniques.
- Prioritize and implement numerous of ideas and hypothesis.
- Document and write a structed report.

### References:

[0] [Kaggle Competition: Personalized Medicine](https://www.kaggle.com/c/msk-redefining-cancer-treatment)
    
[1] [OncoKB: A Precision Oncology Knowledge Base](http://ascopubs.org/doi/full/10.1200/PO.17.00011)

[2] [Cancer-specific High-throughput Annotation of Somatic Mutations: computational prediction of driver missense mutations](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2763410/)

[3] [Predicting the Functional Consequences of Somatic Missense Mutations Found in Tumors](https://www.ncbi.nlm.nih.gov/pubmed/24233781)

[4] [Predicting the functional impact of protein mutations: application to cancer genomic](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3177186/)

[5] [tmVar: A text mining approach for extracting sequence variants in biomedical literature](https://www.ncbi.nlm.nih.gov/research/bionlp/Tools/tmvar/)

[6] [TaggerOne: Joint Named Entity Recognition and Normalization with Semi-Markov Models](https://www.ncbi.nlm.nih.gov/research/bionlp/Tools/taggerone/)

[7] [GNormPlus: An Integrative Approach for Tagging Gene, Gene Family and Protein Domain](https://www.ncbi.nlm.nih.gov/research/bionlp/Tools/gnormplus/)

[8] [ Personalised Medicine - EDA with tidy R]( https://www.kaggle.com/headsortails/personalised-medicine-eda-with-tidy-r)

[9] [Redefining Treatment]( https://www.kaggle.com/the1owl/redefining-treatment-0-57456)

[10] [Brief insight on Genetic variations]( https://www.kaggle.com/dextrousjinx/brief-insight-on-genetic-variations)

[11] [Human Genome Variation Society]( http://www.hgvs.org/)

[12] [Official external data and pre-trained models thread]( https://www.kaggle.com/c/msk-redefining-cancer-treatment/discussion/35810)

[13] [Key Sentences Extraction ideas]( https://github.com/suika-go/suika/blob/master/kernel/baseline.ipynb)

[14] [KAGGLE ENSEMBLING GUIDE]( https://mlwave.com/kaggle-ensembling-guide/)

[15] [Introduction to Ensembling/Stacking in Python]( https://www.kaggle.com/arthurtok/introduction-to-ensembling-stacking-in-python)

[16] [Titanic Top 4% with ensemble modeling]( https://www.kaggle.com/yassineghouzam/titanic-top-4-with-ensemble-modeling)

[17] [Entrez-biopython]( http://biopython.org/DIST/docs/api/Bio.Entrez-module.html)
  
  
